mods/mc_worldmanager/teleporterNode.lua

```diff
@@ -0,0 +1,190 @@
+    else
+        local realm = Realm.GetRealm(tonumber(realmID))
+        local realmName = "unknown"
+        if (realm ~= nil) then
+            realmName = realm.Name
+        end
+        descriptionString = descriptionString .. "(" .. realmName .. ")"
+    end
+
+    return descriptionString
+end
+
+minetest.register_node("mc_worldmanager:teleporter", {
+    description = "Teleporter Node (Spawn)",
+    tiles = { { name = "mc_worldmanager_teleporter.png", color = "white" } },
+    overlay_tiles = { { name = "mc_worldmanager_teleporter_crystal.png" } },
+    palette = "mc_worldmanager_palette.png",
+    paramtype2 = "color",
+    groups = { crumbly = 3, soil = 1 },
+
+    on_rightclick = function(pos, node, clicker, itemstack, pointed_thing)
+
+        local meta = minetest.get_meta(pos)
+
+        local instanced = string.lower(meta:get_string("instanced")) == "true"
+
+        local temp = string.lower(meta:get_string("temp")) == "true"
+
+        local realmID = meta:get_int("realm")
+
+        local realmName = (meta:get_string("name"))
+        local realmSchematic = (meta:get_string("schematic"))
+
+        if (realmSchematic == "") then
+            realmSchematic = nil
+        end
+
+        Debug.log(tostring(realmID) .. " " .. tostring(instanced) .. " " .. tostring(temp) .. " " .. tostring(realmName) .. " " .. tostring(realmSchematic))
+
+
+        -- If our realmID is set to 0 and we're not instanced, teleport to spawn!
+        if (realmName == "spawn") then
+            local spawn = mc_worldManager.GetSpawnRealm()
+            spawn:TeleportPlayer(clicker)
+            return nil
+        end
+
+        -- We now need to figure out where we are teleporting players
+        local realmObject = nil
+
+        -- If we are an instanced realm, we let our realm instancing logic take care of this; Otherwise we will use the realm ID to get the realm.
+        if (instanced == true) then
+            realmObject = mc_worldManager.GetCreateInstancedRealm(realmName, clicker, realmSchematic, temp)
+        else
+            realmObject = Realm.GetRealm(realmID)
+
+            -- If we couldn't find the realm, we will create a new one if there is a schematic associated with this teleporter.
+            if (realmObject == nil) then
+                if (realmSchematic ~= nil) then
+                    realmObject = Realm:NewFromSchematic(realmName, realmSchematic)
+                else
+                    realmObject = Realm:New(realmName, { x = 80, y = 80, z = 80 })
+                    realmObject:CreateGround()
+                    realmObject:CreateBarriersFast()
+                end
+
+                meta:set_int("realm", realmObject.ID)
+                minetest.swap_node(pos, { name = "mc_worldmanager:teleporter", param2 = math.ceil(math.sin(realmObject.ID) * 255) })
+            else
+                minetest.chat_send_player(clicker:get_player_name(), "This teleporter is broken. Please contact an instructor for assistance.")
```

Comment:
Might also be useful to indicate that the student can use the reporting function in the student notebook to report this issue.

Reply:
Good idea, I'll look into this.